src/main/java/com/core/controller/UserController.java

```diff
@@ -0,0 +1,59 @@
+package com.core.controller;
+
+import com.core.model.User;
+import com.core.model.dto.user.UserRequestDto;
+import com.core.model.dto.user.UserResponseDto;
+import com.core.service.RoleService;
+import com.core.service.UserService;
+import com.core.service.mapper.user.UserMapper;
+import java.util.Set;
+import javax.validation.Valid;
+import lombok.AllArgsConstructor;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+@AllArgsConstructor
+public class UserController {
+    private final UserService userService;
+    private final RoleService roleService;
+    private final UserMapper mapper;
```

Comment:
`ModelMapping` is universal therefore could be called just `mapper` as variable while `UserMapper` is specific `User` mapper. If I would be needed to extend this program and add new mapper I would be forced to rename your ambiguous `mapper`.

Reply:
renamed it